### PR TITLE
Fix issues with <dataselect> element

### PR DESCRIPTION
**Pull request by slipher**
In particular, it had a tendency to issue spurious cvar changes when no one was clicking on it.

### CODE REVIEW

File: src/cgame/rocket/rocketDataSelect.h [hunk 1]
```diff
@@ -85,35 +86,36 @@ class RocketDataSelect : public Rml::ElementFormControlDataSelect, public Rml::E
 	{
 		extern std::queue< RocketEvent_t * > eventQueue;
 
-		if ( event.GetTargetElement() == owner && event == "show" )
```
Comment (DolceTriade) [inline src/cgame/rocket/rocketDataSelect.h:88]: Is this guaranteed to not get events from children?

File: pkg/unvanquished_src.dpkdir/ui/server_create.rml [hunk 2]
```diff
@@ -60,7 +60,7 @@
 			</row>
 
 			<row id="maplist">
-				<dataselect source="mapList.default" fields="mapName" valuefield="mapLoadName" name="map" cvar="ui_dialogCvar1"/>
```
Comment (illwieckz) [inline pkg/unvanquished_src.dpkdir/ui/server_create.rml:63]: isn't it a way to remember the selection?

File: src/cgame/rocket/rocketDataSelect.h [hunk 3]
```diff
@@ -85,35 +86,36 @@ class RocketDataSelect : public Rml::ElementFormControlDataSelect, public Rml::E
...
-			eventQueue.push( new RocketEvent_t( Rml::String( va( "setDS %s %s %d", dsName.c_str(), tableName.c_str(), selection ) ) ) );
-
-			// dispatch event so rocket knows about it
-			Rml::Dictionary parameters;
-			parameters[ "index" ] = va( "%d", selection );
-			parameters[ "datasource" ] = dsName;
-			parameters[ "table" ] = tableName;
-			DispatchEvent( "rowselect", parameters );
+			eventQueue.push( new RocketEvent_t( Rml::String(
+				va( "setDS %s %s %d", dsName.c_str(), tableName.c_str(), GetSelection() ) ) ) );
```
Comment (DolceTriade) [inline src/cgame/rocket/rocketDataSelect.h:113]: Rml::String is now stdstring right? So you might as well use Str::Format here?

**Review by DolceTriade — COMMENTED** [review]
When deleting code, please add a rationale to the commit message for posterity.

File: src/cgame/rocket/rocketDataSelect.h (hunk 3 above)
Comment (slipher) [reply src/cgame/rocket/rocketDataSelect.h]: Done.

File: src/cgame/rocket/rocketDataSelect.h (hunk 1 above)
Comment (slipher) [reply src/cgame/rocket/rocketDataSelect.h:88]: Changed it back to if statements.

File: pkg/unvanquished_src.dpkdir/ui/server_create.rml (hunk 2 above)
Comment (slipher) [reply pkg/unvanquished_src.dpkdir/ui/server_create.rml:63]: No, it doesn't do anything. Once this is merged, why don't you finish up #2883? 😛

**Comment by slipher** [comment]
> When deleting code, please add a rationale to the commit message for posterity.

I added one to the nuke language data source commit.

**Review by DolceTriade — APPROVED** [review]

**Pull request merged**